### PR TITLE
Migrate Chord Class from ASCII Tab JS

### PR DESCRIPTION
**Pull request by shadanan**
Migrated the Chord class from ASCII Tab JS. Also converted to ES6. This
will be used for detecting and transposing chords in the Music Markdown.

### CODE REVIEW

File: chord.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  'add2|add4|add9|',
+  'aug|',
+  'dim|dim7|dim11|dim13|',
+  'sus|sus4|sus2|sus2sus4|',
+  '-5',
+  ')?',
+  '(?:/([CDEFGAB](?:#|b)?))?'
+].join('');
+
+const chordPattern = new RegExp('^' + pattern + '$');
```
Comment (vincelugli) [inline chord.js]: Probably better to have these contained in the class as const member variables. Currently, these will all be global variables, which might cause problems at some point.

File: chord.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+class Chord {
+  constructor(root, quality, bass) {
+    this.root = root;
+    this.quality = quality || '';
+    this.bass = bass || '';
+  }
+
+  toString() {
+    let result = this.root + this.quality;
+    if (this.bass != '') {
```
Comment (vincelugli) [inline chord.js]: This can be shortened to `if (this.bass) { ... }` if you wanted it to handle undefined/falsy values.

File: chord.js (hunk 1 above)
Comment (shadanan) [reply chord.js]: Tried to move it in, but got this error on npm test:
`Support for the experimental syntax 'classProperties' isn't currently enabled`

File: chord.js (hunk 2 above)
Comment (shadanan) [reply chord.js]: Ok!

**Comment by vincelugli** [comment]
👍

File: chord.js (hunk 1 above)
Comment (vincelugli) [reply chord.js]: Huh, that's an odd error. I'll take a look at it some, and we can resolve it in the next PR.

**Pull request merged**